test/cppsuite/test_harness/workload/database_operation.cxx

```diff
@@ -161,27 +159,26 @@ database_operation::insert_operation(thread_context *tc)
         /* Collection cursor. */
         auto &cc = ccv[counter];
         while (tc->transaction.active() && tc->running()) {
-            /* Insert a key value pair. */
-            bool rollback_required = tc->insert(cc.cursor, cc.coll.id, start_key + added_count);
-            if (!rollback_required) {
+            /* Insert a key value pair, rolling back the transaction if required. */
+            if (!tc->insert(cc.cursor, cc.coll.id, start_key + added_count)) {
+                added_count = 0;
+                tc->transaction.rollback();
```

Comment:
I am thinking that if absolutely no insert operations are successful, the test will still be successful since:

- There is nothing inserted, hence nothing to check
- Nothing to read
- Nothing to update

Reply:
It could be another ticket but should we allow a maximum number of failures?

Reply:
Yeah this doesn't actually test anything per se it just aims to stress WT and see if it creates a failure internally.